[CS2103-T11-3] Powerlifting Competition Coordinator (P.C.C.)

ooimingsheng opened the pull request:
@shawnkoh @HoWeiChin @Belbel98

ooimingsheng committed:
Add notes file to test commit

ooimingsheng committed:
Update ReadMe Build Status badge

ooimingsheng committed:
Remove codacy badge to test travis enabling auto-publishing of documentation

christabelteo committed:
Update Project Website (#4)

* Update project website

Specifically the AboutUs, ContactUs, and README page.

* Edit AboutUs page

Changed 'prof' to 'professor'
Made changes to responsibilities, and added component ic for everyone.

christabelteo committed:
Tutorial adding command (#9)

* Update .gitignore

* Enhance RemarkCommand

Completed steps up till this step on the tutorial

* Add Remark class to the model

* Add placeholder element for remark to the UI

* Modify Person to support Remark field

* Update Storage

Add Remark field to JsonAdaptedPerson class

* Finalise UI - Connect Remark field to PersonCard

* Modify RemarkCommand execute() method

ooimingsheng committed:
Revert "Tutorial adding command (#9)" (#10)

This reverts commit f7d6ca2480c96df459afd223376593b438f58aba.

ooimingsheng committed:
Edit developer guide (#3)

* Change repo url

* Update developer guide to include user stories, user profile,  non-functional requirements, use cases

* Edit repository URL

* Edit repository URL to correct URL

sorry the previous commit was incorrect

Commit:
User guide for powerlifting coordinator  (#22)

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide.adoc

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-== Use Cases
+== Use Cases (To Do)
+
+(For all use cases below, the *System* is the `Powerlifting Competition Competitor` and the *Actor* is the `user`, unless specified otherwise)
```
The use of the term user can be better defined: e.g. athlete/participant

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*Extensions*
+
+[none]
+* 2a. The list is empty.
```
Perhaps your system can show an error message when the list is empty

ChangUo79 commented on docs/DeveloperGuide.adoc:
```diff
@@ -264,28 +373,152 @@ Certain properties of the application can be controlled (e.g user prefs file loc
+*Value proposition*: manage powerlifting competitions faster than a typical mouse/GUI driven app
 
 [appendix]
 == User Stories
```
The user story format is correct and the 3 parts are all present. The benefit matches the function nicely. Overall everything is in order and there is no missing user stories. Good job!

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*MSS*
+
+1. User obtains a list of competitions by listing (<<Use case: UC1 - View my competition list, *UC1*>>)
+2. User requests for a new powerlifting competition to be deleted.
```
Can cancel the word 'new'

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+1. User obtains a list of competitions by listing (<<Use case: UC1 - View my competition list, *UC1*>>)
+2. User requests for a new powerlifting competition to be deleted.
+3. System adds the new competition to the user's competition list.
```
Should be delete instead of add

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+*MSS*
+
+1. User obtains a list of competitions by listing (<<Use case: UC1 - View my competition list, *UC1*>>)
```
Underline the listing with the UC1 since it is referenced. Also, whenever you reference a use case, the same name has to be used

ChangUo79 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,22 +823,140 @@ Use case ends.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `1.8.0_60` or higher installed.
+.  Should work on both 32-bit and 64-bit environments.
+.  Should come with automated unit tests and open source code.
+.  Should come with a user guide and a developer guide.
```
These NFRs do not have anything to do with the system and they do not specify the constraints under which system is develeoped and operated.

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+* 2a. System does not find any athletes satisfying all the constraints.
++
+[none]
+** 2a1. System shows an empty list.
```
Maybe showing an error message would be clearer for the user instead of an empty list

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+1. User obtains a list of athletes by listing (<<Use case: UC5 - View my athlete list, *UC5*>>).
+or searching (<<Use case: UC6 - Search for athletes,*UC6*>>).
+2. User requests to view more information on a specific athlete in the list.
+3. System shows detailed information of the book.
```
I think it's athlete instead of book

junhuplim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+1. User requests for an athlete to be added.
+2. User inputs athlete data.
+3. User confirms that the addition of the athlete cannot be reversed.
```
Why is this relevant since you can delete a athlete in UC10

damithc commented:
Incorrect PR name corrected. Please take note of your correct team ID ...

Commit:
Set up models (#24)

* Set up models

* Remove attempt and participation models

christabelteo committed:
Participation Component (#26)

* Update .gitignore

to exclude all .iml files

* Create Attempt and Participation packages

* Remove trailing white spaces in UserGuide.adoc

Commit:
Update user guide (#30)

* Ui.png added to UG

* Ui.png added to UG with space

* New Ui.png

* Delete Ui.png

* Add files via upload

ooimingsheng committed:
Abstract out unique lists and exceptions (#32)

ooimingsheng committed:
Create storage for multiple classes (#34)

* Make generic storage

* Add Json classes for participation and competition

christabelteo committed:
Set up Attempt, Exercise, Participation Components (#27)

* Create ObservableList for the different Attempts

* Add Exceptions

* Change Attempt, Exercise, Participation classes

as well as updated changes to review PR

* Optimise imports

* Minor changes

* Add JavaDoc Comment in Participation.java

Commit:
Person class changes propagated (#35)

* Person class changes propagated

Update code to pass checkstyle

Fixed customDate equals method

Change name from String to Name class

Update tests due to changes in Person class

* Update src/main/java/seedu/address/model/person/CustomDate.java

Co-Authored-By: Christabel <50402053+belbel98@users.noreply.github.com>

ooimingsheng committed:
Refactor participation class and include JSON storage for participation class (#38)

* Refactor participation class

* Add participation data storage

christabelteo committed:
Remove unnecessary commands in the User Guide

For the participation feature

christabelteo committed:
Update UserGuide.adoc

Commit:
New comp features (#44)

* New Branch

* CRUD for Competitions

* CRUD added to front end

* Remove trailing whitespace from user guide

ooimingsheng committed:
Add scoring methods (#45)

christabelteo committed:
Participation lists (#43)

* Create list participation command and parser

* Optimise imports for checkstyle

* Add alternate command result

* Add ListPartCommandParserTest

* checkstyle fix

* Add method to get the top attempts in Participation.java

* Added front end to display participation list

* Add empty constructor for ListPartCommand

* Fix attempt creation from json file issue

* Update ListPartCommandParserTest.java

removed tests for the tests which were not working..

* checkstyleTest fix

* Combined methods to getThreeLiftScore()

* Regroup methods in Participation.java

christabelteo committed:
Update Session commands on UG

Commit:
CRUD for Competitions added

christabelteo committed:
Update command for Session Handling

christabelteo committed:
Update UserGuide for correct session handling commands (#47)

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

christabelteo committed:
Session Handling (#48)

* Create Session package

* Update Participation.java

* Update UserGuide.adoc

* Change constructor parameter for Session

* Setup Session component

* Create runtime exceptions

* Add commands to handle session

* Fix session handling

Session can now import participations with preloaded attempts, and can continue the previous session from where they are left off.

* Fix import order for checkstyle

* Add ObservableValue for SessionInformation to be displayed on frontend

* Revert "Add ObservableValue for SessionInformation to be displayed on frontend"

This reverts commit 9a91c3ac2b3b5da44561dfeb2793a66450ff02ed.

* Allow session to display following lifter and attempt

* Remove LoadAttemptsCommand, add AddPartCommand

attempt submission will be required when creating new participation.

* Update Model and ModelManager

* Fix checkstyle and optimise imports

* Optimise imports

Commit:
New comp features (#49)

* New Branch

* CRUD for Competitions

* CRUD added to front end

* UserGuide.adoc fixed

* checks start/end dates

* Ranking added

* Ranking modified

* Tests issues resolved

* additional changes made as per pr comments

* rebased and test bug fixes

christabelteo committed:
Update session command result, create package for competition in logic (#50)

ooimingsheng committed:
Scope commands to in/out of session (#51)

ooimingsheng committed:
Refactor file structure and names (#53)

ooimingsheng committed:
Add show hide abilities (#54)

Commit:
Add overall rank command (#56)

ooimingsheng committed:
Change ReadMe Ui pic (#57)

ooimingsheng committed:
Make ranklist command (#59)

Commit:
Change UI and other minor adjustments (#60)

* New Branch

* CRUD for Competitions

* CRUD added to front end

* UserGuide.adoc fixed

* checks start/end dates

* additional changes made as per pr comments

* UI changes

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Delete Ui.png

Commit:
Add files via upload

Latest Ui.png

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Latest updates.

Commit:
Update UserGuide.adoc

christabelteo committed:
Made changes to session and participation commands (#61)

* Made changes to session and participation commands

* Optimise imports

* Update UserGuide on new/edited commands

* Fix trailing whitespaces

ooimingsheng committed:
Fix some dry PE issues (#111)

ooimingsheng committed:
Fix more dry PE issues (#112)

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update UserGuide.adoc

Commit:
Update build.gradle

Commit:
Update build.gradle

christabelteo committed:
Fix session related issues (#116)

* Fix issue #106

Wrong Error Message when listParticipation competition name does not exist

* Fix issue #104

Allows negative values to be entered for the weight carried, and also the space between kg and weight.

* Fix issue #82

During competition, doesn't show who comes next.

* Fix issues #81 , #98

* Fix issue #102

No indicator when the competition has ended and who won.

* Refactor in/out session command/parser files into correct package

* add prefix_comp for startSession command

* Fix issue #100

* move file to correct package

* Reorder imports

Commit:
Resolve dry pe (#117)

* resolve ranking and issue 65

* Further furnishing for issue 65

* further refinements

* further refinements

* Powerlifting Icon added

Commit:
Update build.gradle

Commit:
Update build.gradle

Commit:
Update build.gradle

Commit:
Update build.gradle

ooimingsheng committed:
Change Ui.PNG extension to .png (#119)

christabelteo committed:
Update Developer Guide (incomplete) (#120)

* Remove unnecessary sections in DG

* Added participation feature into DG

and other miscellenous (adding diagrams)

* Add endSession feature description

christabelteo committed:
Minor fix on NextLifterCommand (#122)


* Update NextLifterCommand.java

Commit:
Authorship issues (#125)

* resolve ranking and issue 65

* Further furnishing for issue 65

* further refinements

* further refinements

* Powerlifting Icon added

* UG issues resolved

* add authorship

* authorship

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

christabelteo committed:
Fix trailing whitespace for travis build

christabelteo committed:
Added session features and sequence diagrams (#126)

ooimingsheng committed:
Make tests (#127)

LGTM

ooimingsheng committed:
Add PPP for Ming Sheng

Commit:
Dg (#128)

* DG updated

* DG updated

* removed uml for rank feature

* more refinements

Commit:
Add search participation and search comp (#123)

ooimingsheng committed:
Modify user guide, developer guid and PPP (#129)

Commit:
Dg (#130)

* DG updated

* DG updated

* removed uml for rank feature

* more refinements

* PPP and DG updated

* AddressBook changed to system

* diagram changed

* AddressBookParser to SystemParser

ooimingsheng committed:
WIP: Update diagrams in developer guide (#131)

* Add documentation to developer guide

* Add more documentation to the developer guide

Commit:
Update AboutUs.adoc

Commit:
Update AboutUs.adoc

Commit:
Rename HoWeiChin.adoc to howeichin.adoc

Commit:
Update AboutUs.adoc

Commit:
Update UserGuide.adoc

ooimingsheng committed:
Modify PPP for Ming (#133)

Commit:
User guide, developer guide and project portfolio page (#134)

ooimingsheng committed:
Adjust PPP for ming sheng (#135)

christabelteo committed:
Add PPP for Christabel (#136)

Commit:
Add documentation for find participation and find competition commands (#137)

christabelteo committed:
Update Christabel's PPP

ooimingsheng committed:
Fix links on PPP (#138)

christabelteo committed:
Fix tagging issues on Christabel's PPP

christabelteo committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T11-3/main

christabelteo committed:
Change name to full name

and other minor changes

christabelteo committed:
Delete unnecessary section on DG

christabelteo committed:
Bug fix on startSession (#139)

* Bug Fix

Bug fix for starting competition session not in filtered competition list

* Fix deleteParticipation result error

ooimingsheng committed:
Fix bug in ranklist command parser (#140)
